sdk/core/azure-core/test/ut/curl_connection_pool_test.cpp

```diff
@@ -601,6 +601,30 @@ namespace Azure { namespace Core { namespace Test {
       }
     }
 
+    TEST(CurlConnectionPool, forceConnectionClosed)
```

Comment:
Can we have a test that resembles the real-life scenario that this PR is fixing, a bit more?

I don't follow how the PR is fixing the issue, other than returning a `CURL_SEND_ERROR` in one case, in `SendBuffer`.

Reply:
I believe that this test should *exactly* reflect the real-life scenario that this PR is fixing. We call into the azuresdkforcpp test server asking for a 102 response, which also includes a Connection: close response. Before my fix, the call to Perform would have failed, after the fix, the call to Perform succeeds and returns headers indicating that the connection was closed.

Could you help me understand why it doesn't reflect the real-life scenario?

Reply:
@ahsonkhan, I am going to resolve this comment thread now so that I can merge this PR.
If you'd like more follow up your question, please create a work item so that we can follow up on it later.